### PR TITLE
Make URL_REGEX more strict

### PR DESCRIPTION
**Pull request by SomeoneToIgnore**
Current URL_REGEX matches things like `http::trace::on_request::log_parameters` which are not valid URLs, the patch fixes this, by requiring `//` after all protocols that should have it.

### CODE REVIEW

**Comment by SomeoneToIgnore** [comment]
Thank you for the man pages pointer, I've also updated the dev guide to mention this bit.

I don't have strong intentions to merge this, feel free to close this if you want to.
Originally, I copied this regex from this repo since I wanted to match regex in other terminal and thought that this regex will get me covered when matching selections on hover — and when found out that there's only a small bit missing, decided to upstream this.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -86,7 +86,7 @@ Latency is another important factor for Alacritty. On X11, Windows, and macOS th
 Code should be documented where appropriate. The existing code can be used as a guidance here and
 the general `rustfmt` rules can be followed for formatting.
 
-If any change has been made to the `config.rs` file, it should also be documented in the man pages.
+If any change has been made to the `config.rs`, `ui_config.rs` files, it should also be documented in the man pages.
```
Comment (kchibisov) [inline CONTRIBUTING.md:89]: I'm not sure that this should be changed that way.  And I don't think it's a part of this patch in the first place.

Comment (SomeoneToIgnore) [reply CONTRIBUTING.md]: Reverted.

**Review by kchibisov — APPROVED** [review]

**Comment by kchibisov** [comment]
I'd still wait for @chrisduerr to ack this one though.

File: alacritty/src/config/ui_config.rs [hunk 2]
```diff
@@ -25,7 +25,7 @@ use crate::config::window::WindowConfig;
 
 /// Regex used for the default URL hint.
 #[rustfmt::skip]
-const URL_REGEX: &str = "(ipfs:|ipns:|magnet:|mailto:|gemini:|gopher:|https:|http:|news:|file:|git:|ssh:|ftp:)\
+const URL_REGEX: &str = "(ipfs:|ipns:|magnet:|mailto:|gemini://|gopher://|https://|http://|news:|file://|git://|ssh:|ftp://)\
```
Comment (chrisduerr) [inline alacritty/src/config/ui_config.rs:28]: I'm not sure `file://` should be enforced. I wouldn't be surprised to see `file:/home/user/whatever` in the wild and it generally works just fine.

File: alacritty/src/config/ui_config.rs [hunk 3]
```diff
@@ -547,3 +547,60 @@ impl PartialEq for LazyRegexVariant {
...
 }
 impl Eq for LazyRegexVariant {}
+
+#[cfg(test)]
+mod tests {
+    use alacritty_terminal::term::test::mock_term;
+
+    use crate::display::hint::visible_regex_match_iter;
+
+    use super::*;
```
Comment (chrisduerr) [inline alacritty/src/config/ui_config.rs:557]: Super should be the first import.

**Review by chrisduerr — CHANGES_REQUESTED** [review]

File: alacritty/src/config/ui_config.rs (hunk 2 above)
Comment (SomeoneToIgnore) [reply alacritty/src/config/ui_config.rs]: Sure, done.

File: alacritty/src/config/ui_config.rs (hunk 3 above)
Comment (SomeoneToIgnore) [reply alacritty/src/config/ui_config.rs]: Fixed.

**Review by chrisduerr — APPROVED** [review]

**Pull request merged**